### ISSUE TITLE
BackAndroid.exitApp() prevent to hide Toast message

### ISSUE THREAD

**Issue by exromany** [facebook/react-native#5599]
0.18.0

When I show Toast message and invoke BackAndroid.exitApp() while it still shown, Toast message will not be closed.

```js
ToastAndroid.show('This message will not be closed', ToastAndroid.SHORT);
setTimeout(() => {
  BackAndroid.exitApp();
}, 1000);
```

**Comment by hufeng** [comment]
yes, i met the same problem.

**Comment by juncocoa** [comment]
yes, i met the same problem.

**Comment by AndrewJack** [comment]
#5906 Looks like it will help fix this issue.

**Comment by suxiaoyu** [comment]
yes, i met the same problem.

**Comment by naivehhr** [comment]
+1

**Comment by nihgwu** [comment]
the same

**Comment by tdzl2003** [comment]
It looks like react-native changes the thread to run native modules. This causes toast message cannot disappear after exitApp(). I'll give a PR today.

**Comment by janicduplessis** [comment]
@tdzl2003 Thanks for submitting the PR! I'll close this issue since it will be fixed once #6443 lands.

**Comment by janicduplessis** [comment]
@facebook-github-bot close
